irs/adapters/outbound/kafka_producer.py

```diff
@@ -14,90 +14,104 @@
+from pydantic import BaseSettings, Field
 
+from irs.ports.outbound.event_pub import EventPublisherPort
 
-class UploadValidationSuccessEvent(BaseModel):
-    """Contains details for successfully checksum validated file processing"""
-
-    file_id: str = Field(..., alias="file-id")
-    secret_id: str = Field(..., alias="secret-id")
-    offset: int
-    part_size: int = Field(..., alias="part-size")
-    part_checksums_md5: list[str] = Field(..., alias="part-checksums-md5")
-    part_checksums_sha256: list[str] = Field(..., alias="part-checksums-sha256")
-    content_checksum_sha256: str = Field(..., alias="sha256-checksum")
-
-    class Config:
-        """Needed config to allow popupalation by non-alias name"""
-
-        allow_population_by_field_name = True
-
-
-class UploadValidationFailureEvent(BaseModel):
-    """Contains details in case of file checksum validation failure"""
-
-    file_id: str = Field(..., alias="file-id")
-    cause: str
-
-    class Config:
-        """Needed config to allow popupalation by non-alias name"""
-
-        allow_population_by_field_name = True
 
+class EventPubTanslatorConfig(BaseSettings):
+    """Config for publishing file upload-related events."""
 
-async def get_publisher() -> KafkaEventPublisher:
-    """Factoring this out makes it overridable by tests"""
-    return KafkaEventPublisher.construct(config=CONFIG)
-
-
-async def produce_success_event(
-    file_id: str,
-    secret_id: str,
-    offset: int,
-    part_size: int,
-    part_checksums_md5: list[str],
-    part_checksums_sha256: list[str],
-    content_checksum_sha256: str,
-):  # pylint: disable=too-many-arguments
-    """Produce and send an event if checksum validation was successful"""
-
-    type_ = "upload_validation_success"
-    data = UploadValidationSuccessEvent(
-        file_id=file_id,
-        secret_id=secret_id,
-        offset=offset,
-        part_size=part_size,
-        part_checksums_md5=part_checksums_md5,
-        part_checksums_sha256=part_checksums_sha256,
-        content_checksum_sha256=content_checksum_sha256,
-    ).dict()
-    publisher = await get_publisher()
-    await publisher.publish(
-        payload=data,
-        type_=type_,
-        key=file_id,
-        topic=CONFIG.topic,
+    files_to_register_success_type: str = Field(
+        "file_validation_success",
+        description=(
+            "The type used for events informing about the success of a file validation."
+        ),
+        example="file_validation_success",
+    )
+    files_to_register_failure_type: str = Field(
+        "file_validation_failure",
+        description=(
+            "The type used for events informing about the failure of a file validation."
+        ),
+        example="file_validation_failure",
+    )
+    files_to_register_topic: str = Field(
+        "file_interrogation",
+        description=(
+            "The topics used for events informing about the outcome of file validations."
```

Comment:
```suggestion
            "Name of the topic used for events informing about the outcome of file validations."
```
Don't forget to run update_config_docs afterwards